src/v/cluster/rm_stm.cc

```diff
@@ -2665,4 +2665,15 @@ ss::future<> rm_stm::clear_old_tx_pids() {
       });
 }
 
+ss::future<> rm_stm::clear_old_idempotent_pids() {
+    while (_log_state.lru_idempotent_pids.size() > 100500) {
```

Comment:
100500? let's name and document this, and even better, use a config for it?

Reply:
nit: it's easier to review if we introduce and and wire `max_concurrent_producer_ids` as the first commits of the PR. In that case we don't need to use the 100500 placeholder